code/modules/unit_tests/inhands.dm

```diff
@@ -0,0 +1,91 @@
+/// Makes sure items with defined inhand_icon_states... actually have icons that exist!
+/datum/unit_test/defined_inhand_icon_states
+	var/static/list/possible_icon_states = list()
+	var/fallback_log_message
+	var/unset_inhand_var_message
+
+/datum/unit_test/defined_inhand_icon_states/proc/generate_possible_icon_states_list(directory_path)
+	if(!directory_path)
+		directory_path = "icons/mob/inhands/"
+	for(var/file_path in flist(directory_path))
+		if(findtext(file_path, ".dmi"))
+			for(var/sprite_icon in icon_states("[directory_path][file_path]", 1)) //2nd arg = 1 enables 64x64+ icon support, otherwise you'll end up with "sword0_1" instead of "sword"
+				possible_icon_states[sprite_icon] += list("[directory_path][file_path]")
+		else
+			possible_icon_states += generate_possible_icon_states_list("[directory_path][file_path]")
+
+/datum/unit_test/defined_inhand_icon_states/Run()
+	generate_possible_icon_states_list()
+
```

Comment:
I clarified in https://github.com/TaleStation/TaleStation/pull/2783 , but to add your modular files to this unit test, add this line underneath the other generate_possible_icon_states_list()
`generate_possible_icon_states_list("talestation_modules/icons/mob/inhands/")`

then update deploy.sh w/ the same path.